### PR TITLE
feat: introduced home layout

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

- This PR introduces numerous changes on CodeBox, Shikiji, CodeTabs in order to make it functional for requirements for the Home Page
- This PR introduces the Home Layout and all related changes for making all feature-set of the Home Page to work on its fullest
- This PR does a lot of accessibility fixes related to the Home Page and related Components
- There are a few code cleanups to related components and parts of the codebase

### Additional Changes

- Link component types were updated to better fit Anchor Element types
- Button Component was changed to be an anchor element with role of button since this fits better the case of many of ours buttons actually being links
- Fixed CSS specifity of mdxContent as changing it from a class name "mdxContent" to be a html tag "main"

Keep in mind this PR adds a few `todos` of things that must be cleaned up once we remove the legacy website design

### Screenshot

image

### Preview

Branch Preview: [nodejs-org-git-feat-codebox-and-home-reqs-openjs.vercel.app/en](https://nodejs-org-git-feat-codebox-and-home-reqs-openjs.vercel.app/en)

## Validation

Home Page should work on local deployments with `npm run serve:redesign`

### CODE REVIEW

File: next.rewrites.mjs [hunk 1]
```diff
@@ -37,12 +38,23 @@ const redirects = async () => {
...
+  const mappedRewrites = siteRedirects.internal.map(
+    ({ source, destination }) => ({
+      source: source.replace('/:locale', localesMatch),
       destination,
-    })),
-  };
+    })
+  );
+
+  // This allows us to remape legacy website URLs to the temporary redesign ones
```
Comment (bmuenzenmeyer) [inline next.rewrites.mjs]: typo

```suggestion
  // This allows us to remap legacy website URLs to the temporary redesign ones
```

File: public/static/images/dark-backdrop.svg
Comment (bmuenzenmeyer) [inline public/static/images/dark-backdrop.svg:1]: i thought we had these already in the storybook - and had hoped to use them as part of blog previews OG - but I could be mistaken

**Review by bmuenzenmeyer — COMMENTED** [review]
a small nit and a comment

also, going to https://nodejs-12wt6n3bo-openjs.vercel.app/en and then clicking on the node.js logo in the top right renders this:

image

File: public/static/images/dark-backdrop.svg
Comment (ovflowd) [reply public/static/images/dark-backdrop.svg:1]: Well, feel free to find these SVGs and replace in this PR. Maintainers are welcome to make changes :)

**Comment by ovflowd** [comment]
> also, going to [nodejs-12wt6n3bo-openjs.vercel.app/en](https://nodejs-12wt6n3bo-openjs.vercel.app/en) and then clicking on the node.js logo in the top right renders this:

Completely normal. The redesign page is actually under /en/new-design, the reason why you can see when opening /en/ through an URL is due to a URL rewrite.

File: public/static/images/dark-backdrop.svg
Comment (ovflowd) [reply public/static/images/dark-backdrop.svg:1]: Doing it

**Comment by ovflowd** [comment]
Branch Preview: https://nodejs-org-git-feat-codebox-and-home-reqs-openjs.vercel.app/en

File: components/Downloads/DownloadButton.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const nodeDownloadLink = downloadUrlByOS(versionWithPrefix, os, bitness);
+
+  const onDownloadNode = () => push(nodeDownloadLink);
+
+  return (
+    <>
+      <Button
+        variant="special"
+        onClick={onDownloadNode}
+        className="hidden flex-row items-center justify-center gap-2 dark:flex"
```
Comment (mikeesto) [inline components/Downloads/DownloadButton.tsx]: Curious why these tailwind classes (and the ones below) are inline? I thought we were trying to use CSS modules

Comment (ovflowd) [reply components/Downloads/DownloadButton.tsx]: True. I was experimenting here, but these should def go to CSS modules

File: components/Downloads/DownloadButton/index.module.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.downloadButton {
+  @apply flex-row
+    items-center
+    justify-center
+    gap-2;
+
+  svg {
+    @apply h-[20px]
+      w-[20px]
```
Comment (AugustinMauroy) [inline components/Downloads/DownloadButton/index.module.css]:
```suggestion
    @apply size-5
```
[docs ref](https://tailwindcss.com/docs/size)
_just verify if we are on lasted tailwind_

File: components/Common/Button/index.module.css [hunk 4]
```diff
@@ -70,11 +68,9 @@
        z-[-1]
        h-full
        w-full
```
Comment (AugustinMauroy) [inline components/Common/Button/index.module.css]:
```suggestion
       size-full
```

File: components/Downloads/DownloadButton/index.module.css (hunk 3 above)
Comment (ovflowd) [reply components/Downloads/DownloadButton/index.module.css]: Nice!

**Comment by ovflowd** [comment]
@canerakdas we might also want to look the accessibility and best practices report from Lighthouse and see why it is complaining

**Comment by canerakdas** [comment]
Hey @ovflowd ,

- The text in the title area is cut off due to the background clip. Honestly, I do not know how it can be solved. Wouldn't it be enough to use text color in this area with the correct contrast ratio, like the paragraph below it?

  - image

- IMO we need to review the variant naming of the buttons. The button we specified as secondary should be a ghost and it looks like we need a neutral button styling. We must use the neutral version of the Codebox component, as in Figma. This component is used in many places and I think we do not want these buttons to be positioned as primary action on the pages (I can open a separate PR for this)

  - image

- When I look at the low accessibility score, there are generally missing aria-labels and the texts in the codebox component do not have the correct contrast ratio. IMO the Codebox component is critical because we are using this everywhere :) I can work on the current theme coloring to reduce this contrast ratio to at least the [minimum value](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) 👀 

  - https://shikiji.netlify.app/guide/theme-colors

**Comment by ovflowd** [comment]
> The text in the title area is cut off due to the background clip. Honestly, I do not know how it can be solved. Wouldn't it be enough to use text color in this area with the correct contrast ratio, like the paragraph below it?

I think it's fine. It's just a minor detail.

> When I look at the low accessibility score, there are generally missing aria-labels and the texts in the codebox component do not have the correct contrast ratio. IMO the Codebox component is critical because we are using this everywhere :) I can work on the current theme coloring to reduce this contrast ratio to at least the [minimum value](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) 👀

We would need to modify the theme we use, (simply copy the json file and put it under out public folder or whatnot) But for the aria-label, I thought all our components were already accessible because of Radix, what components are missing aria-labels? And can you add them if you don't mind? (Remember we need to add to i18n en.json whatever that needs to be translated)

**Comment by ovflowd** [comment]
> IMO we need to review the variant naming of the buttons. The button we specified as secondary should be a ghost and it looks like we need a neutral button styling. We must use the neutral version of the Codebox component, as in Figma. This component is used in many places and I think we do not want these buttons to be positioned as primary action on the pages (I can open a separate PR for this)

Not sure I understood your statements here correctly 🤔 what exactly is wrong with the secondary variant?

File: components/Downloads/DownloadButton/index.module.css [hunk 5]
```diff
@@ -5,8 +5,8 @@
     gap-2;
 
   svg {
-    @apply h-[20px]
-      w-[20px]
+    @apply h-5
```
Comment (ovflowd) [inline components/Downloads/DownloadButton/index.module.css]: Shouldn't it be `size-5` as @AugustinMauroy mentioned?

File: layouts/New/layouts.module.css [hunk 6]
```diff
@@ -86,7 +86,7 @@
   .hexagonBackdrop {
     @apply absolute
       left-0
-      z-[-1]
+      -z-[1]
```
Comment (ovflowd) [inline layouts/New/layouts.module.css]: Shouldn't it be just -z-1?

File: pages/en/new-design/index.mdx [hunk 7]
```diff
@@ -6,12 +6,15 @@ layout: home.hbs
 
 <WithBadge section="index" />
 
+<div className="flex flex-col gap-4">
```
Comment (ovflowd) [inline pages/en/new-design/index.mdx]: Can we avoid adding this div? Reasoning is the less code we add it here, the easier it is for the translators

File: layouts/New/layouts.module.css (hunk 6 above)
Comment (canerakdas) [reply layouts/New/layouts.module.css]: In Tailwind `z-index` information can be multiples of 10 / or manually, I think there is no harm in saying 10 here. I will update it 👍

**Comment by canerakdas** [comment]
> Not sure I understood your statements here correctly 🤔 what exactly is wrong with the secondary variant?

IMO it may be useful to find a common word for prop names and values and update them.

In the current situation;

| Component | Values                                         | Prop    |
|-----------|------------------------------------------------|---------|
| Button    | `'primary' \| 'secondary' \| 'special'`          | `variant` |
| Badge     | `'default' \| 'warning' \| 'error'`              | `kind`    |
| Banner    | `'default' \| 'warning' \| 'error'`              | `type`    |
| Preview   | `'announcement' \| 'release' \| 'vulnerability'` | `type`    |

I think having a standard for these will make the design system more consistent and easier to use by everyone. Our primary color may be green, but it doesn't seem right to call the transparent version a secondary. While we named other components according to their color values 🤔

**Comment by ovflowd** [comment]
> > Not sure I understood your statements here correctly 🤔 what exactly is wrong with the secondary variant?
> 
> IMO it may be useful to find a common word for prop names and values and update them.
> 
> In the current situation;
> 
> Component	Values	Prop
> Button	`'primary' | 'secondary' | 'special'`	`variant`
> Badge	`'default' | 'warning' | 'error'`	`kind`
> Banner	`'default' | 'warning' | 'error'`	`type`
> Preview	`'announcement' | 'release' | 'vulnerability'`	`type`
> I think having a standard for these will make the design system more consistent and easier to use by everyone. Our primary color may be green, but it doesn't seem right to call the transparent version a secondary. While we named other components according to their color values 🤔

Let's use "type" as common word

**Comment by ovflowd** [comment]
Also FYI I've done a bunch of a11y fixes so the accessibility score should be higher up now. We still need to improve colour scheme of the CodeBox. The Color Scheme comes from shikiji.config.js

**Comment by ovflowd** [comment]
@canerakdas it'd be great if you added an inline doc explaining the code replacement and why it was done

**Comment by ovflowd** [comment]
cc @nodejs/nodejs-website I think this PR is ready for review. Can we get reviews?

**Review by mikeesto — APPROVED** [review]
LGTM, nice work

**Review by bmuenzenmeyer — APPROVED** [review]

**Review by AugustinMauroy — APPROVED** [review]
Oh wow a big job has been done here.

**Pull request merged**